core/ast.h

```diff
@@ -317,6 +318,15 @@ struct Index : public AST {
     { }
 };
 
+struct Slice : public AST {
```

Comment:
Basically you could extend Index to add the last and step ASTs which are null if they're not provided.